src/plugins/slurm/slurm-spank-auks.c

```diff
@@ -501,51 +482,68 @@ spank_auks_remote_init (spank_t sp, int ac, char *av[])
 		goto exit;
 	}
 
-	/* get user credential */
-	fstatus = auks_api_get_cred(&engine,uid,auks_credcache);
-	if ( fstatus != AUKS_SUCCESS ) {
-		xerror("unable to get user %u cred : %s",uid,
-		      auks_strerror(fstatus));
-		unlink(auks_credcache);
-		auks_credcache[0]='\0';
+	/* Get auks cred */
+	fstatus = auks_api_get_auks_cred(&engine,uid,&cred);
+	if( fstatus ) {
+		xerror("unable to unpack auks cred from reply : %s",
+		       auks_strerror(fstatus));
+		fstatus = AUKS_ERROR_API_CORRUPTED_REPLY ;
 		goto unload;
 	}
-	xinfo("user '%u' cred stored in %s",uid,auks_credcache);
 
-	/* change file owner */
-	fstatus = chown(auks_credcache,uid,gid);
-	if ( fstatus != 0 ) {
-		xerror("unable to change cred %s owner : %s",
-		      auks_credcache,strerror(errno));
+	/* change to user uid and gid before getting cred */
+	if ( setegid(gid) ) {
+		xerror("unable to switch to user gid : %s",
+		       strerror(errno));
 		goto unload;
 	}
 
-	/* set cred cache name in user env */
-	fstatus = spank_setenv(sp,"KRB5CCNAME",auks_credcache,1);
-	if ( fstatus != 0 ) {
-		xerror("unable to set KRB5CCNAME env var");
+	if ( seteuid(uid) ) {
+		xerror("unable to switch to user uid : %s",
+		       strerror(errno));
+		goto unload;
 	}
```

Comment:
setegid+seteuid is not sufficient to avoid some security issues. proper handling of secondary groups using initgroups is necessary.

Reply:
See https://github.com/hautreux/auks/pull/38#discussion_r425643871